### ISSUE TITLE
ContextMenu muda issue

### ISSUE THREAD

**Issue by ogoffart** [slint-ui/slint#9031]
### Bug Description

Can be reproduced by right clicking on a line edit in the gallery on Windows:

 - When compiled with rust: The position of the popup is not correct
 - When using the interpreter: The menu doesn't show at all.
 - (When compiling with C++: it doesn't use native menu, but that's acceptable for now)

### Environment Details

- Slint Version: master branch (1.13)
- Platform/OS: Windows  (because it needs muda)
- Backend/Renderer: Winit

**Comment by ogoffart** [comment]
CC @npwoods

**Comment by npwoods** [comment]
The issue seems to be that I'm expecting the `position` passed to `show_native_popup_menu()` to be in window coordinates, whereas the coordinates seem to actually be relative to the control in question.  IMHO this the correct behavior, because there is no context about the "current" control passed in.

I tried running the below example on Slintpad (which lacks `muda`), and the context menus definitely seem to be relative to the control that was originally clicked.  This is definitely not what I would expect.

In my application I do something similar to what the example below does, but the context menu is triggered by a callback from my Rust code, so in practice it behaves how I would expect (absolute coordinates)

It isn't clear to me how this should be resolved
````slint
import { Button, HorizontalBox, LineEdit, VerticalBox } from "std-widgets.slint";

export component MainWindow inherits Window {
    width: 600px;
    height: 600px;
    context-menu := ContextMenuArea {
        Menu {
            MenuItem {
                title: "Hydrogen";
            }

            MenuItem {
                title: "Helium";
            }

            MenuItem {
                title: "Lithium";
            }
        }
    }

    VerticalBox {
        Button {
            text: "Alpha";
            clicked => {
                context-menu.show({ x: self.absolute-position.x, y: self.absolute-position.y + self.height });
            }
        }

        LineEdit {
            text: "Bravo";
        }

        LineEdit {
            text: "Charlie";
        }

        Button {
            text: "Delta";
            clicked => {
                context-menu.show({ x: self.absolute-position.x, y: self.absolute-position.y + self.height });
            }
        }
    }
}
````

**Comment by ogoffart** [comment]
For the non-muda case, the absolute position is computed there:
https://github.com/slint-ui/slint/blob/351324ae292e1896e3abe20547a11ff15be64c76/internal/core/window.rs#L1185-L1186

**Comment by npwoods** [comment]
@ogoffart I'm curious if you approve of the approach I took in #9084; if you approve it I can try my hand at fixing the CI failures

**Comment by tronical** [comment]
With #9084, the gallery compiled with Rust now has a well working context menu. But `cargo run slint-viewer -- examples\gallery\gallery.slint` still yields a context menu that's invisible.

**Comment by ogoffart** [comment]
I'm working on some simplification that should fix this.

Edit: done in https://github.com/slint-ui/slint/pull/9159
